File: Lehrverwaltung/client/src/app/core/entities/dtos/ExerciseDTO.ts
import {ProgrammingLanguages} from '../../models/types/TYPES';
import {FileObject} from '../../models/file/FileObject';

export class ExerciseDTO {
  title: string;
  description: string;
  language: ProgrammingLanguages;
  requirement: string;
  links: string[];
  privateInput: string;
  privateOutput: string;
  publicInput: string;
  publicOutput: string;
  codeFile: FileObject;


  constructor(title: string, description: string, language: ProgrammingLanguages, requirement: string, links: string[], privateInput: string, privateOutput: string, publicInput: string, publicOutput: string, codeFile: FileObject) {
    this.title = title;
    this.description = description;
    this.language = language;
    this.requirement = requirement;
    this.links = links;
    this.privateInput = privateInput;
    this.privateOutput = privateOutput;
    this.publicInput = publicInput;
    this.publicOutput = publicOutput;
    this.codeFile = codeFile;
  }
}

File: Lehrverwaltung/server/src/main/java/tie/lehrlinge/lehrverwaltung/server/model/dto/ExerciseDTO.java
package tie.lehrlinge.lehrverwaltung.server.model.dto;

import java.util.List;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ExerciseDTO {

  private String title;
  private String language;
  private String requirement;
  private String description;
  private List<String> links;
  private String privateInput;
  private String publicInput;
  private String publicOutput;
  private String privateOutput;
  private FileObjectDTO codeFile;
}
